pkg/apis/autoscaling/v1alpha1/metricsproducer.go

```diff
@@ -46,17 +46,47 @@ type PendingCapacitySpec struct {
 	NodeSelector map[string]string `json:"nodeSelector"`
 }
 
-type ScheduledCapacitySpec struct {
-	// NodeSelector specifies a node group. The selector must uniquely identify a set of nodes.
-	NodeSelector map[string]string `json:"nodeSelector"`
+type Timezone string
+
+type ScheduleSpec struct {
 	// Behaviors may be layered to achieve complex scheduling autoscaling logic
 	Behaviors []ScheduledBehavior `json:"behaviors"`
+	// Defaults to UTC. Users will specify their schedules assuming this is their timezone
+	// ref: https://en.wikipedia.org/wiki/List_of_tz_database_time_zones
+	// +optional
+	Timezone *Timezone `json:"timezone,omitempty"`
+	// A schedule defaults to this value when no behaviors are active
+	DefaultReplicas int32 `json:"defaultReplicas"`
 }
 
 // ScheduledBehavior defines a crontab which sets the metric to a specific replica value on a schedule.
 type ScheduledBehavior struct {
-	Crontab  string `json:"crontab"`
-	Replicas int32  `json:"replicas"`
+	// The value the MetricsProducer will emit when the current time is within start and end
+	Replicas int32       `json:"replicas"`
+	Start    *TimeConfig `json:"start"`
+	End      *TimeConfig `json:"end"`
+}
+
+type TimeConfig struct {
+	// Specify only one of Crontab or Pattern
+	Crontab *string  `json:"crontab,omitempty"`
+	Pattern *Pattern `json:"pattern,omitempty"`
+}
+
+// List of 3-letter abbreviations i.e. "Jan, Feb, Mar"
+type Months string
+
+// List of 3-letter abbreviations i.e. "Mon, Tue, Wed"
+type Weekdays string
+
+// Pattern is a strongly-typed version of crontabs
```

Comment:
Can you add doc on what it means if a field is omitted? Also what are the rules about which fields can or cannot be specified?